### DISCUSSION TITLE
Cypress.env present but saying it is undefined

### DISCUSSION THREAD

**Discussion by diogormendes** [cypress-io/cypress#25087] [Questions and Help] [answered]
I'm saving an API tokes as a `Cypress.env` 

<img width="371" alt="Screenshot 2022-12-09 at 11 04 29" src="https://user-images.githubusercontent.com/17742979/206720493-2c406d21-91a1-4433-a7dd-b26a612ca756.png">

When I call this on the test, it says is undefined. 
But printing the envs right before we can see it is there.
<img width="425" alt="Screenshot 2022-12-09 at 11 04 42" src="https://user-images.githubusercontent.com/17742979/206720648-bc5ac457-33b8-48cc-8215-eaa515a989fe.png">
<img width="379" alt="Screenshot 2022-12-09 at 11 03 40" src="https://user-images.githubusercontent.com/17742979/206720656-df154a28-5b44-4b16-82e2-ce3f7dc4215e.png">

What may I be doing wrong ?

**Comment by emilyrohrbough** [comment, marked as answer]
@diogormendes This likely how your test is written given how this value is resolving at the time these commands are being queued.

If you have:
```js
.then(() => {
  Cypress.env('token', 'val')
})

cy.log(Cypress.env('token'))
```
The `Cypress.env('token')` value at the time `cy.log()` is added indeed undefined.

If you were add the cy.log() in a then callback, it would be be queued after the env value has been set, it will be defined:
```js
.then(() => {
  Cypress.env('token', 'val')
})
.then(() => {
  cy.log(Cypress.env('token'))
})
```
